Create Parallelize.py

andrzeng opened the pull request:
This is a uncommented draft version of Parallelize.py. It has been successfully tested to run masking/unmasking operations on the 1001-subject dataset. However, more testing is needed in order to streamline and massage the code

andrzeng committed:
Create Parallelize.py

This is a uncommented draft version of Parallelize.py. More testing is needed in order to streamline and massage the code

andrzeng committed:
Update Parallelize.py

steelec commented:
@andrzeng ,
Given that this is a draft should we wait until it has been a bit more tested? If you document its usage in the README.md and add it to the __init__ on the current branch we can get others to test as well before merging into the master.

steelec commented:
from what i see here, this is specific to processing calls that return only numerics that are castable to `np.array`?

andrzeng commented:
> @andrzeng , Given that this is a draft should we wait until it has been a bit more tested? If you document its usage in the README.md and add it to the **init** on the current branch we can get others to test as well before merging into the master.

Yes!! I didn't intend for this to be merged immediately. I created the pull request so you could view the code and we could share comments about it. I will add some comments and write a short blurb about it in the README file before letting others test it out





> from what i see here, this is specific to processing calls that return only numerics that are castable to `np.array`?

The f() function does not have to return an array-like structure. I used simple functions like this one during testing:
https://i.imgur.com/YdqFQ2L.png
also, the runtime is significantly higher than expected due to the time taken to initialize multiple processes

andrzeng committed:
Added comments, used a "hack" to improve runtime

Added comments to the code. Used a global variable hack to drastically improve runtime (an amygdala masking task for 1001 subjects dropped from 3 minutes to 19 seconds)

andrzeng committed:
Update __init__.py

andrzeng committed:
Added example usage for Parallelize()

andrzeng committed:
Update README.md
